_posts/2016-11-02-improving-page-speed-with-graphql.markdown

```diff
@@ -0,0 +1,113 @@
+---
+layout: post
+title: "Improving Page Speed with GraphQL"
+date: 2016-11-02 14:21
+author: kana
+categories: [graphql, javascript, positron, joi]
+---
+
+This past year, our team started using a GraphQL orchestration layer that connects various APIs with multiple front-end apps including [iOS](http://artsy.github.io/blog/2016/06/19/graphql-for-mobile). It also handles caching and extracts some business logic out of our client apps. This helped us not only to be more consistent with the way we fetch data across apps, but also improved developer happiness and even bridged teams by having our web and iOS developers work with the same API layer. This got me thinking what other problems GraphQL could solve at Artsy.
+
+I work on the Publishing Team at Artsy, and we've recently been focused on page speed as a KPI. With so many ways of measuring speed, it's a daunting task but for this post, I'll focus on the way we handled things on the server-side and how integrating GraphQL on our API improved page speed.
+
+<!-- more -->
+
+## Prior to GraphQL
+
+Before we went ahead with adding a GraphQL-based endpoint in Positron, I spent about a week and refactoring our current codebase. We refactored our router code to make less fetches, made better use of caching, and moved some below-the-fold rendering into the client side. These are important steps towards a faster page, but it doesn't push our technology in a direction that lets us reimagine things.
```

Comment:
It might also be helpful to call out the name "Artsy Writer" and include a screenshot/gif to give some product perspective. That, or we could include screenshots in Positron's README like [Eigen](https://github.com/artsy/eigen) does.